src/main/java/com/everis/control/CarService.java

```diff
@@ -15,17 +18,23 @@
 
 	@Inject
 	private PersistenceService<Car, String> persistenceService;
+	
+	private Map<String, String> filterMap = new HashMap<String, String>();
 
 	/**
 	 * Method to get a list of Car Entity available
 	 */
 
-	public List<CarDto> getCars() {
-		List<CarDto> cars=this.persistenceService.getEntitiesWithNamedQuery("Car.findAll", Car.class).stream().
-				map(car -> car.mapToDto()).collect(Collectors.toList());
-		return cars;
+	public List<CarDto> getCars(int page, int size, String filterBy, String orderBy) {
+		
+		filterMap.put("brand", filterBy);
+		filterMap.put("country", filterBy);
+		filterMap.put("registration", filterBy);
+		TypedQuery<Car> query = this.persistenceService.getEntitiesQuery(Car.class, filterMap, orderBy);
+		query.setFirstResult((size * page) - size);
```

Comment:
Usually it is not a good practice to mix business logic with query logic. Please move this query related code to the persistence service, so everything query related is in the same place.